### DISCUSSION TITLE
Error 500 blank page when admin add or edit product with +40.000 categories.

### DISCUSSION THREAD

**Discussion by viv3s** [PrestaShop/PrestaShop#40171] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

I have a page with a lot of data in version 1.6 is working.
It has more than 40,000 categories.
Now developing a new shop in version 1.7.5 I can't enter to edit products, I get a blank page (500 error) even when debug mode is activated. If I move the categories from level depth 6 to id_shop = 2, it leaves me with 30,000 categories then it works, but if I move them again to id_shop = 1 again it fails again.

I have a dedicated server and with really good performance.

I think the new category selector in the admin product page is not optimized. 
¿Can you check it?

Thank you.

![2022-05-20 11_17_28-Lil Nas X - THATS WHAT I WANT](https://user-images.githubusercontent.com/11992127/169498755-a4de55e8-2d6d-4f51-b2af-b08115d0f6b9.png)

### Steps to reproduce

1. Create a fresh prestashop installation, 
2. create more than 40.000 categories (6level depth) (i suggest to do it via webservice),
3. acces to edit product page.

### PrestaShop version(s) where the bug happened

1.7.3+

### PHP version(s) where the bug happened

7.2

**Comment by viv3s** [comment]
I had it on 1.7.3 and I upgraded to 1.7.5 first and then 1.7.6 same error happens.

**Comment by hibatallahAouadni** [comment]
Hello @viv3s 

Your issue is related to #12907
And as we decide that every performance bug will be treated as an improvement.
So the Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank for your understanding :pray:
